src/Guard.php

```diff
@@ -88,19 +88,7 @@ public function __construct(
             throw new RuntimeException('CSRF middleware failed. Minimum strength is 16.');
         }
         $this->strength = $strength;
-        if (is_array($storage)) {
-            $this->storage = &$storage;
-        } elseif ($storage instanceof ArrayAccess) {
-            $this->storage = $storage;
-        } else {
-            if (!isset($_SESSION)) {
-                throw new RuntimeException('CSRF middleware failed. Session not found.');
-            }
-            if (!array_key_exists($prefix, $_SESSION)) {
-                $_SESSION[$prefix] = [];
-            }
-            $this->storage = &$_SESSION[$prefix];
-        }
+        $this->storage = &$storage;
```

Comment:
If the check is only for the `$_SESSION` in `validateStorage` we will have two places to set `$this->storage`. I think it is better to keep it in a single method. Also passing by reference twice is not an issue in the case of `ArrayAccess`.